packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -305,6 +310,33 @@ export const useGridFilter = (apiRef: GridApiRef, rowsProp: GridRowsProp): void
     }
   }, [apiRef, clearFilteredRows, logger, rowsProp]);
 
+  const prevFilterModelRef = React.useRef();
+
+  React.useEffect(() => {
+    return apiRef.current.subscribeEvent(GRID_FILTER_MODEL_CHANGE, (params) => {
+      const prevFilterModel = prevFilterModelRef.current;
+      prevFilterModelRef.current = params.filterModel;
+
+      const trimFilterModel = (filterModel) => ({
+        ...filterModel,
+        items: filterModel.items.filter((item) => item.value !== undefined),
+      });
+
+      if (
+        !prevFilterModel ||
+        !isDeepEqual(trimFilterModel(params.filterModel), trimFilterModel(prevFilterModel))
+      ) {
+        apiRef.current.applyFilters();
+      }
+    });
+  }, [apiRef]);
+
+  React.useEffect(() => {
+    return apiRef.current.subscribeEvent(GRID_ROWS_UPDATED, () => {
+      apiRef.current.applyFilters();
+    });
+  }, [apiRef]);
```

Comment:
It seems that this was missing, we didn't explicitly depend on the row stream update API.

Reply:
Not necessary it seems, it was working before. Check /story/x-grid-tests-streaming--single-subscription-fast

Reply:
My bad, I don't know how to read. It was already there, at the top of the file.